source/adapters/opencl/kernel.cpp

```diff
@@ -9,6 +9,8 @@
 //===----------------------------------------------------------------------===//
 #include "common.hpp"
 
+#include <memory>
```

Comment:
what's this for? did including `<regex>` in common.hpp pull this in before it was removed?

Reply:
yes, the build was not seeing `shared_ptr` any more, I think that was because of regex was including this.